projects/plugins/jetpack/tests/e2e/specs/post-connection/search-dashboard.test.js

```diff
@@ -0,0 +1,67 @@
+import { test, expect } from '../../fixtures/base-test.js';
+import { SearchDashboard } from 'jetpack-e2e-commons/pages/wp-admin/index.js';
+import { enableInstantSearch, disableInstantSearch } from '../../helpers/search-helper.js';
+import { prerequisitesBuilder, Plans } from 'jetpack-e2e-commons/env/index.js';
+import playwrightConfig from '../../playwright.config.cjs';
+
+test.describe( 'Search', () => {
+	let searchDashboard;
+
+	test.beforeAll( async ( { browser } ) => {
+		const page = await browser.newPage( playwrightConfig.use );
+		await prerequisitesBuilder( page )
+			.withLoggedIn( true )
+			.withConnection( true )
+			.withPlan( Plans.Complete )
+			.withActiveModules( [ 'search' ] )
+			.build();
+
+		await enableInstantSearch();
+		await page.close();
+	} );
+
+	test.afterAll( async () => {
+		await disableInstantSearch();
+	} );
+
+	test.beforeEach( async ( { page } ) => {
+		searchDashboard = await SearchDashboard.visit( page );
+		await searchDashboard.waitForNetworkIdle();
+	} );
+
+	test( 'Can display dashboard correctly', async () => {
+		expect( await searchDashboard.isSearchModuleToggleVisibile() ).toBeTruthy();
+		expect( await searchDashboard.isInstantSearchToggleVisible() ).toBeTruthy();
+		expect( await searchDashboard.isTitleVisible() ).toBeTruthy();
+		expect( await searchDashboard.isHeaderVisible() ).toBeTruthy();
+		expect( await searchDashboard.isFooterVisible() ).toBeTruthy();
+		expect( await searchDashboard.isCustomizeButtonVisible() ).toBeTruthy();
+		expect( await searchDashboard.isEditWidgetButtonVisible() ).toBeTruthy();
+	} );
+
+	test( 'Can toggle search module and instant search', async () => {
+		// When toggling off search module, instant search is toggled off too.
+		await searchDashboard.toggleSearchModule();
+		expect( await searchDashboard.isSearchModuleToggleOn() ).toBeFalsy();
+		expect( await searchDashboard.isInstantSearchToggleOn() ).toBeFalsy();
```

Comment:
Should we also wait for the toggles to become re-enabled during these tests?

Reply:
Good catch. The test would wait for the saving request to be finished however it didn't check whether the toggles would stick. I added another waiting condition to make sure the toggles are not toggling in 8c022218e6